geomesa-filter/src/main/scala/org/locationtech/geomesa/filter/visitor/QueryPlanFilterVisitor.scala

```diff
@@ -316,6 +318,16 @@ class QueryPlanFilterVisitor(sft: SimpleFeatureType) extends DuplicatingFilterVi
     }
   }
 
+  override def visit(filter: PropertyIsLike, extraData: Any): AnyRef = {
+    try {
+      new LikeToRegexConverter(filter).getPattern
```

Comment:
do you still need to pass it into `Pattern.compile`? looking at the LikeToRegexConverter I don't think it throws any validation errors, but I could be wrong.

Reply:
Good question; doing this was sufficient to throw an exception in the provided unit test.  I think that exception may have been in the GeoTools code which was trying to parse / extract the regex?

Reply:
I'll toss in the call to Pattern.compile.